### PR TITLE
Reduce repeated discard_shreds checks

### PR DESCRIPTION
**Pull request by steviez**
#### Summary of Changes
Removing a repeated check as mentioned in https://github.com/solana-labs/solana/pull/26225#discussion_r909033829

### CODE REVIEW

File: ledger/src/shred.rs [hunk 1]
```diff
@@ -691,7 +691,7 @@ pub fn should_discard_shred(
     };
     let slot = match layout::get_slot(shred) {
         Some(slot) => {
-            if slot <= root || !slot_bounds.contains(&slot) {
```
Comment (steviez) [inline ledger/src/shred.rs:694]: Coding shreds - moved this check to later on once we know what kind of shred we have
Data shreds - this is later checked by calling `verify_shred_slots()`
https://github.com/solana-labs/solana/blob/36d13af015c2830530dd68faae770a558aee0839/ledger/src/blockstore.rs#L4029-L4036

`root <= parent && parent < slot` ==> `root <= parent < slot` ==> `root < slot` ==> `slot > root`. In old logic, we error if `slot <= root` so it must be that `slot > root`

File: ledger/src/shred.rs [hunk 2]
```diff
@@ -733,10 +737,6 @@ pub fn should_discard_shred(
                     return true;
                 }
             };
-            if parent_offset == 0 && slot != 0 {
-                stats.bad_parent_offset += 1;
-                return true;
-            }
```
Comment (steviez) [inline ledger/src/shred.rs:739]: In `verify_shred_slots()`, we only allow `parent_offset == 0` if `slot == 0` so this is redundant.

File: ledger/src/blockstore.rs [hunk 3]
```diff
@@ -4026,7 +4026,7 @@ macro_rules! create_new_tmp_ledger_fifo_auto_delete {
     };
 }
 
-pub(crate) fn verify_shred_slots(slot: Slot, parent: Slot, root: Slot) -> bool {
+pub fn verify_shred_slots(slot: Slot, parent: Slot, root: Slot) -> bool {
```
Comment (behzadnouri) [inline ledger/src/blockstore.rs:4029]: why make this `pub`?

File: ledger/src/blockstore.rs [hunk 4]
```diff
@@ -5885,6 +5885,21 @@ pub mod tests {
         }
     }
 
+    #[test]
+    pub fn test_verify_shred_slots() {
```
Comment (behzadnouri) [inline ledger/src/blockstore.rs:5889]: not `pub`.

**Review by behzadnouri — APPROVED** [review]

File: ledger/src/blockstore.rs (hunk 3 above)
Comment (steviez) [reply ledger/src/blockstore.rs:4029]: Thank you - no clue why I changed this

**Comment by steviez** [comment]
Ninja'd myself and had to rebase, sanity succeeded so going to merge as I previously got through CI with all greens

**Pull request merged**